### PR TITLE
Organiser and User profile page

### LINKED ISSUE #5: US 01.02.02
As an entrant I want to update information such as name, email and contact information on my profile

### LINKED ISSUE #6: US 01.03.01
As an entrant I want to upload a profile picture for a more personalized experience

### LINKED ISSUE #8: US 01.03.03
As an entrant I want my profile picture to be deterministically generated from my profile name if I haven't uploaded a profile image yet.

### PR DESCRIPTION
**Pull request by ghost**
This is most of the functionality for the organiser profile and probably all of the functionality for the user/entrant profile minus server connectivity.

Tests, UML and such to be completed still.

To see different spawned Avatars, write different text in the Name text box before hitting delete and you will be given a different avatar. To use the upload button, follow the screen to google cloud and you can log into your google account where you will be able to access any photos you have stored in google photos.

To switch between user and organiser profile pages, go into navigation > user_mobile_navigation.xml and switch out the android:name field to the name of the fragment you want to see.

Also prompts popup that asks user permission to send notifications

### CODE REVIEW

**Comment by SuperCreativeUniqueUsername** [comment]
I don't have time to look over the code tonight, but I thought I would add the few things I noticed on my first impression. This pr doesn't seem to be connected to the project board (and the issues that this pr addresses aren't explicitly stated). I think moving forward it would be a good idea to branch off specific issues on the project board. For now, if you could just link this pr to the related issues that would be great.

**Comment by SuperCreativeUniqueUsername** [comment]
This is something else I was thinking about, but I wasn't sure how to phrase it properly. I understand that you're halfway done with what you wanted to do (adding or changing facility profile photos). but i just wanted to say I think it's important for pull requests to meet the standards we've outlined. Last meeting we talked about PR requests being more throughout. Which would mean testing code, and updating the UML, CRC, and storyboards. etc etc.  And I don't want to come across like I'm bossing you around or telling you what to do, but I do think its important that we try to meet those standards. And being that this PR is so massive I think its even more important to make it very clear what its doing and ensure that CRC, UML, storyboards are updated so that nothing is missed.

**Comment by ghost** [comment]
@SuperCreativeUniqueUsername the plan was to complete those things before I created the pull request... you said writing tests would make it harder to review so I'm not exactly sure what to do? The other parts I will get to when I have the time, just not tonight.

File: app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java
Comment (Kori-Kozicki) [inline app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java:279]: What do setting these onClickListeners do? Do they like generally call on click cause that would be handy

**Review by Kori-Kozicki — APPROVED** [review]
I think overall this looks really good! A few suggestions for the future is that maybe we should take away the option to even choose the admin role (or at least for accounts without admin permissions). Also I think the automated profile pictures are nice but how related are they to the names? I tried a few tests and I can see the relation but it's not the most clear. Might be something to touch base with Alex about. Oh and also how up to date is this with main? becuase it would be good to have that before a merge.

File: app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java
Comment (ghost) [reply app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java:279]: This will run the code that opens the photopicker

Comment (ghost) [reply app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java:279]: The profile picture will change depending on what you enter into the name field. The automatic avatar generation is mostly until I place minimum length requirements on the name field. Currently it will create something without you entering anything. 

This one is 5 commits behind main, the one on my computer is current. But I haven't committed again to this branch.

**Comment by SuperCreativeUniqueUsername** [comment]
> @SuperCreativeUniqueUsername the plan was to complete those things before I created the pull request... you said writing tests would make it harder to review so I'm not exactly sure what to do?

Oopsie yeah, I did say that. What I had initially meant by saying adding more would make it harder to review was adding more user stories/features. I realize now I didn't make that clear at all when I wrote that message. Sorry about that. I think that this not having tests is okay for now, and I'll take a look at the code tomorrow. I also still think that refreshing uml, crc, storyboards would be important to include in this pr to avoid headaches later.

File: app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java
Comment (Kori-Kozicki) [reply app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java:279]: Oh so does the name need to be "Updated" before you can see the relation?

**Comment by ghost** [comment]
> > @SuperCreativeUniqueUsername the plan was to complete those things before I created the pull request... you said writing tests would make it harder to review so I'm not exactly sure what to do?
> 
> Oopsie yeah, I did say that. What I had initially meant by saying adding more would make it harder to review was adding more user stories/features. I realize now I didn't make that clear at all when I wrote that message. Sorry about that. I think that this not having tests is okay for now, and I'll take a look at the code tomorrow. I also still think that refreshing uml, crc, storyboards would be important to include in this pr to avoid headaches later.

No problem lol. When I have a bit more time I can go through the UML and other parts.

File: app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java
Comment (ghost) [reply app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java:279]: Yes, updating the name will force a new avatar to be created. If the name box is empty it generates for a string that says null. Either the program or the website caches the avatar so it will not generate a new one with the same name.

Comment (Kori-Kozicki) [reply app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java:279]: So same names will have the same avatar?

Comment (ghost) [reply app/src/main/java/com/example/pygmyhippo/organizer/ProfileFragment.java:279]: If there are exactly the same then yes, but this does fit the user story. 

"As an entrant I want my profile picture to be deterministically generated from my profile name if I haven't uploaded a profile image yet."

I am not 100% clear on whether or not an avatar should be created when a user deletes their profile picture. I intend to ask that today. Edit: Prof said yes to the avatar being generated when the user deletes their uploaded avatar.

File: app/build.gradle.kts
Comment (SuperCreativeUniqueUsername) [inline app/build.gradle.kts:39]: What is this package you are removing?

File: app/src/main/AndroidManifest.xml
Comment (SuperCreativeUniqueUsername) [inline app/src/main/AndroidManifest.xml]: what does this line do?

File: app/build.gradle.kts
Comment (ghost) [reply app/build.gradle.kts:39]: materials design, I added it when I was trying to figure out rounded corners for my first pull request and never used it

File: app/src/main/AndroidManifest.xml
Comment (ghost) [reply app/src/main/AndroidManifest.xml]: Line 20 allows a user to be able to pan the view... like using the hand tool in figma. Essentially using your finger to scroll, or using tab on the keyboard to move the view to the next input so it's not blocked by the keyboard

File: app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java
Comment (SuperCreativeUniqueUsername) [inline app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java]: what does the f mean?

Comment (SuperCreativeUniqueUsername) [reply app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java]: I'm also curious why you choose to use snake case for these variables

File: app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java
Comment (SuperCreativeUniqueUsername) [inline app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java]: I think having more descriptive names for these listeners could make sense.

File: app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java
Comment (ghost) [reply app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java]: f for field

File: app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java
Comment (SuperCreativeUniqueUsername) [inline app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java]: This isn't the most descriptive name for this method. It isn't clear that this method generates an avatar based on a name just based on the name of this method.

File: app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java
Comment (SuperCreativeUniqueUsername) [reply app/src/main/java/com/example/pygmyhippo/user/ProfileFragment.java]: I think this comment can be applied to upload, delete, edit, update listeners

File: app/src/main/res/layout/organiser_fragment_profile.xml
Comment (SuperCreativeUniqueUsername) [inline app/src/main/res/layout/organiser_fragment_profile.xml:136]: My initial impression about using O_P_name for the textVew and O_P_textName for the editText is that this convention isn't very intuitive. When I saw this I thought that the O_P_textName would have been the textView.

File: app/src/main/res/layout/organiser_fragment_profile.xml
Comment (SuperCreativeUniqueUsername) [inline app/src/main/res/layout/organiser_fragment_profile.xml]: It seems like the naming convention for ids changed here to have "profile" instead of using "p". For consistency, it would make sense to change the earlier ones to match with using the full word.

File: app/src/main/res/layout/organiser_fragment_profile.xml
Comment (SuperCreativeUniqueUsername) [inline app/src/main/res/layout/organiser_fragment_profile.xml]: This id changes back to the old convention

Comment (ghost) [reply app/src/main/res/layout/organiser_fragment_profile.xml]: This is not a new id

File: app/src/main/res/layout/user_fragment_profile.xml
Comment (SuperCreativeUniqueUsername) [inline app/src/main/res/layout/user_fragment_profile.xml:1]: The naming convention in this file seems consistent within the file, but it's not consistent with the organiser version.

File: app/src/main/res/layout/organiser_fragment_profile.xml
Comment (ghost) [reply app/src/main/res/layout/organiser_fragment_profile.xml:136]: I added Label to the end of the Textviews

Comment (SuperCreativeUniqueUsername) [reply app/src/main/res/layout/organiser_fragment_profile.xml:136]: Looks great

**Comment by SuperCreativeUniqueUsername** [comment]
Everything looks great, with UML and CRC I think this is ready to be merged

**Comment by katzhong** [comment]
would you also change the storyboard on figma with whatever you have changed here? (ex. i think you changed how editing/deleting a picture would look)

**Comment by ghost** [comment]
> would you also change the storyboard on figma with whatever you have changed here? (ex. i think you changed how editing/deleting a picture would look)

Storyboard is Updated (I didn't change the photos on the Wiki, but the storyboards in Figma), need to to do UML and CRC after class

**Comment by ghost** [comment]
> Everything looks great, with UML and CRC I think this is ready to be merged

UML and CRC are updated

**Review by SuperCreativeUniqueUsername — APPROVED** [review]
looks good

**Pull request merged**
